File: frontend/human-cloud-app/src/services/profileService.js
import RestDataSource from "./restdatasource";

export const profileService = {
    fetchProfileDetails,
    addProfileDetails,
    updateProfileDetails,
    deleteProfileDetails
};

//fetch profile details
function fetchProfileDetails(userId,fn) {

          var url = "http://localhost:8005/humanCloud-identity/api/profile/getProfileDetails?userId="+userId;
         
          return new RestDataSource(url, null, fn).GetData((res) => fn(res));
}

//add profile details
function addProfileDetails(userData, fn) {

    var url = "http://localhost:8005/humanCloud-identity/api/profile/addProfileDetails";
   
    return new RestDataSource(url, 0, fn).Store(userData, (res) => fn(res));
}

//update profile details
function updateProfileDetails(userData, fn) {

    var url = "http://localhost:8005/humanCloud-identity/api/profile/updateProfileDetails";

    return new RestDataSource(url, null, fn).Update(userData, (res) => fn(res));
}

//delete profile details
function deleteProfileDetails(userId, fn) {

    var url = "http://localhost:8005/humanCloud-identity/api/profile/deleteProfile?userId="+userId;
    return new RestDataSource(url, null, fn).Delete(userId, (res) => fn(res));
}

File: backend/identity/identity/src/main/java/com/humancloud/identity/service/ProfileService.java
package com.humancloud.identity.service;

import javax.servlet.http.HttpServletRequest;

import org.springframework.stereotype.Service;

import com.humancloud.identity.entity.User;
import com.humancloud.identity.util.BaseResponse;

@Service
public interface ProfileService {

	BaseResponse<User> addProfileDetails(HttpServletRequest request, User user);

	BaseResponse<User> updateProfileDetails(HttpServletRequest request, User user);

	BaseResponse<User> getProfileDetails(HttpServletRequest request, Long userId);

	BaseResponse<User> deleteProfile(HttpServletRequest request, Long userId);

}
